python/tchannel/tornado/hyperbahn.py

```diff
@@ -58,34 +120,33 @@ def advertise(tchannel, service, routers):
         # TChannel already knows about some of the routers.
         tchannel.peers.get(router)
 
-    @tornado.gen.coroutine
-    def _register():
-        response = yield tchannel.request(service='hyperbahn').send(
-            arg1='ad',  # advertise
-            arg2='',
-            arg3=json.dumps({
-                'services': [
-                    {
-                        'serviceName': service,
-                        'cost': 0,
-                    }
-                ]
-            }),
-            headers={'as': 'json'},
-        )
-
-        if response.code not in ErrorCode:
-            # re-register every ``DEFAULT_TTL`` seconds
-            tornado.ioloop.IOLoop.current().call_later(
-                delay=DEFAULT_TTL,
-                callback=_register,
+    # first advertise rule apply here.
+    ad_future = Future()
```

Comment:
We don't need this `ad_future`, our coroutine's future should be sufficient.

```
attempts = 0
while True:
    response = yield _advertise(tchannel, service)
    if response and response.code is StatusCode.ok:
        break
    attempt_counter += 1
    delay_time = random.uniform(...)
    yield gen.sleep(delay_time)

tornado.ioloop.PeriodicCallback(
    lambda: _advertise(tchannel, service), 
    DEFAULT_DELAY,
).start()
```

For the 30s timeout we can schedule a callback in 30 seconds to `sys.exit` and then cancel it when the `while` loop breaks.